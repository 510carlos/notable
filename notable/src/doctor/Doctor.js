import React, { useEffect, useState } from 'react';
import { getApointment } from './Doctors.helpers';
import { Apointments, Information } from './components';

const Doctor = ({match}) => {
    const id = match.params.id || -1;

    const [apt, setApt] = useState([]);

    useEffect( () => {
        let mounted = true;
        getApointment(id).then( result => {
        const { data } = result;
        if (mounted)
            setApt(data)
        });
        return () => {
            mounted = false;
        }
    }, [id]);

    return (
        <>
            <Information id={id} />
            <Apointments apt={apt} />
        </>
    )
}

export default Doctor;