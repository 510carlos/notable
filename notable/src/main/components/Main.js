import React from 'react';

import 'antd/dist/antd.css';
import { Layout } from 'antd';
import { DoctorContext, DoctorList, DoctorRoutes, GetDoctorsList } from '../../doctor';

const { Content, Sider } = Layout;



const Main = () => {

  const doctorsList = GetDoctorsList();

  return (
    <DoctorContext.Provider value={doctorsList}>
      <Layout>
        <Sider width={200} style={{ background: '#fff' }}>
          <DoctorList />
        </Sider>
        <Layout style={{ padding: '0 24px 24px' }}>
          <Content
            style={{
              background: '#fff',
              padding: 24,
              margin: 0,
              minHeight: 280,
            }}
          >
            <DoctorRoutes />
          </Content>
        </Layout>
      </Layout>
      </DoctorContext.Provider>
  )
};

export default Main;
