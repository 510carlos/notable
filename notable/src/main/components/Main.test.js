import * as React from 'react';

import { shallow } from 'enzyme';
import enzymeToJson from 'enzyme-to-json';

import Main from './Main';

import * as DoctorContext from '../../doctor/Doctor.context';

const dom = <Main />;

const contextValues = [{"id":1,"name":"Hibbert, Julius","email":"kriger@notablehealth.com"},{"id":2,"name":"Krieger, Algemap","email":"kriger@notablehealth.com"},{"id":3,"name":"Riviera, Nick","email":"kriger@notablehealth.com"}];
jest.spyOn(DoctorContext, 'useDoctorContext')
    .mockImplementation(() => contextValues);

describe('Unit tests for Information', () => {
    it('Renders', () => {
        shallow(dom)
    })

    it('Renders with snapshot', () => {
        const wrapper = shallow(dom);
        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    })
})