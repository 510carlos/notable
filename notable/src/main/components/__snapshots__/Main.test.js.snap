// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for Information Renders with snapshot 1`] = `
<ContextProvider
  value={Array []}
>
  <Adapter>
    <Sider
      style={
        Object {
          "background": "#fff",
        }
      }
      width={200}
    >
      <DoctorList />
    </Sider>
    <Adapter
      style={
        Object {
          "padding": "0 24px 24px",
        }
      }
    >
      <Adapter
        style={
          Object {
            "background": "#fff",
            "margin": 0,
            "minHeight": 280,
            "padding": 24,
          }
        }
      >
        <withRouter(DoctorsRoutes) />
      </Adapter>
    </Adapter>
  </Adapter>
</ContextProvider>
`;
